### ISSUE TITLE
out = (SeekableByteChannel) NIOUtils.writableFileChannel(path); is getting null

### ISSUE THREAD

**Issue by RozinaDarediya** [jcodec/jcodec#279]
I am converting my bitmaps into a  video. But **SeekableByteChannel** is null.
I have also given permission in manifest.

<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>

Here is my code : 
<
File file = new File(Environment.getExternalStorageDirectory()+"output.mp4");
        String path = file.getPath();

        SeekableByteChannel out = null;
        try {
            out = (SeekableByteChannel) NIOUtils.writableFileChannel(path); //not understanding this `line`
        } catch (FileNotFoundException e) {
            e.printStackTrace();
            Log.e("msg",e.toString());
        }
        try {
             encoder = new AndroidSequenceEncoder((org.jcodec.common.io.SeekableByteChannel) out, Rational.R(25,1));
        } catch (IOException e) {
            e.printStackTrace();
        }

        Toast.makeText(this, "in for loop", Toast.LENGTH_SHORT).show();
        for(int i=0 ;i<arrayList.size() ; i++){
            try {
                encoder.encodeImage(arrayList.get(i));
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
        Toast.makeText(this, "out of loop", Toast.LENGTH_SHORT).show();
        try {
            encoder.finish();
        } catch (IOException e) {
            e.printStackTrace();
        }
        NIOUtils.closeQuietly(out);>
Hope you will give me sollution very quickly.
Thank you in advance.

**Comment by RozinaDarediya** [comment]
Hi @svitvitskiy 

I got success in solving the above error. Video file is generated but playing video it shows 

> An unknown error occurred

And also getting the error:
`Caused by :  java.lang.IllegalArgumentException: Component 1 width should be a multiple of 2 for colorspace: YUV420J`

Here is my updated MainActivity.java file:
`import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.RequiresApi;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.Toast;

import org.jcodec.api.android.AndroidSequenceEncoder;
import org.jcodec.common.io.NIOUtils;
import org.jcodec.common.io.SeekableByteChannel;
import org.jcodec.common.model.Rational;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.ArrayList;

public class Main2Activity extends AppCompatActivity {

    SeekableByteChannel out = null;
    AndroidSequenceEncoder encoder;
    @RequiresApi(api = Build.VERSION_CODES.N)

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        ArrayList<Bitmap> arrayList = new ArrayList<>();
        Bitmap b = BitmapFactory.decodeResource(getResources(),R.drawable.download);
        Bitmap a = BitmapFactory.decodeResource(getResources(),R.drawable.download1);
        Bitmap c = BitmapFactory.decodeResource(getResources(),R.drawable.download2);

        arrayList.add(a);
        arrayList.add(b);
        arrayList.add(c);


       // File file = new File(Environment.getExternalStorageDirectory()+"output.mp4");
        //String path = file.getPath();


        File file = new File(getExternalFilesDir(null), File.separator+"newDir");
        if(!file.exists()){
            file.mkdirs();
        }
        File file2 = new File(file,"output2.mp4");

        String path = file2.getPath();
        try {
            out =  NIOUtils.writableFileChannel(path);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
            Log.e("msg",e.toString());
        }

        try {
             encoder = new AndroidSequenceEncoder(out, Rational.R(25,1));
        } catch (IOException e) {
            e.printStackTrace();
        }

        Toast.makeText(this, "in for loop", Toast.LENGTH_SHORT).show();
        for(int i=0 ;i<arrayList.size() ; i++){
            try {
                encoder.encodeImage(arrayList.get(i));
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
        Toast.makeText(this, "out of loop", Toast.LENGTH_SHORT).show();
        try {
            encoder.finish();
        } catch (IOException e) {
            e.printStackTrace();
        }
        NIOUtils.closeQuietly(out);
        Toast.makeText(this, "finished", Toast.LENGTH_SHORT).show();
        finish();
    }
}
`
Hope you will respond me quickly

**Comment by svitvitskiy** [comment]
Yes, your video dimensions should be multiple of 2. So make sure that the
video width and height are even.

Stan

**Comment by vrsrohit** [comment]
@RozinaDarediya  hey i am also getting the same error 
did u resolved it ??

**Comment by a1266143** [comment]
hey guys,I got this issue too,any solutions?
